picasso/src/main/java/com/squareup/picasso/Picasso.java

```diff
@@ -596,10 +596,15 @@ private void cancelExistingRequest(Object target) {
 
     @Override public void run() {
       Process.setThreadPriority(THREAD_PRIORITY_BACKGROUND);
+      // Don't inline message. It's important to make sure we keep no local ref.
+      Message message = null;
```

Comment:
I'm trying to repro what I saw in terms of heapdump, and then nullify it, hoping it would help prevent it.

Honestly I'm not really a fan, I just don't know what else to do.